apps/chat/src/components/Marketplace/ApplicationDetails/ApplicationFooter.tsx

```diff
@@ -138,145 +156,338 @@ export const ApplicationDetailsFooter = ({
     [setIsOpenLogs],
   );
 
-  const PlayerIcon = useMemo(() => {
-    switch (playerStatus) {
-      case SimpleApplicationStatus.DEPLOY:
-        return IconPlayerPlay;
-      case SimpleApplicationStatus.UNDEPLOY:
-        return IconPlaystationSquare;
-      case SimpleApplicationStatus.UPDATING:
-      default:
-        return Loader;
-    }
-  }, [playerStatus]);
-
-  const handleUpdateFunctionStatus = () => {
+  const StatusIcons = {
+    [SimpleApplicationStatus.DEPLOY]: IconPlayerPlay,
+    [SimpleApplicationStatus.UNDEPLOY]: IconPlaystationSquare,
+  };
+  const PlayerIcon =
+    playerStatus === SimpleApplicationStatus.UPDATING
+      ? Loader
+      : StatusIcons[playerStatus];
+  const PlayerContextIcon =
+    playerStatus === SimpleApplicationStatus.UPDATING
+      ? LoaderIcon
+      : StatusIcons[playerStatus];
+
+  const handleUpdateFunctionStatus = useCallback(() => {
     dispatch(
       ApplicationActions.startUpdatingFunctionStatus({
         id: entity.id,
         status: getApplicationNextStatus(entity),
       }),
     );
-  };
+  }, [dispatch, entity]);
 
   const handleOpenUnshare = useCallback(
     () => dispatch(ShareActions.setUnshareEntity(entity)),
     [dispatch, entity],
   );
 
+  const handleOpenSharing = useCallback(() => {
+    dispatch(
+      ShareActions.share({
+        featureType: FeatureType.Application,
+        resourceId: entity.id,
+      }),
+    );
+  }, [dispatch, entity.id]);
+
   const isApplicationsSharingEnabled = useAppSelector((state) =>
     SettingsSelectors.isFeatureEnabled(state, Feature.ApplicationsSharing),
   );
 
+  const link = useMemo(
+    () =>
+      `${window.location.origin}/${PageType.Marketplace}?${MarketplaceQueryParams.model}=${entity.reference}`,
+    [entity.reference],
+  );
+
+  const handleCopy = useCallback(() => {
+    if (!navigator.clipboard) return;
+    navigator.clipboard.writeText(link);
+    dispatch(UIActions.showSuccessToast(t('Link copied!')));
+  }, [dispatch, link, t]);
+
+  const menuItems: DisplayMenuItemProps[] = useMemo(
+    () => [
+      {
+        name: t('Copy link'),
+        dataQa: 'application-copy-link',
+        display: isPublicApp && isSmallScreen,
+        Icon: IconLink,
+        onClick: (e: React.MouseEvent) => {
+          handleCopy();
+          e.preventDefault();
+          e.stopPropagation();
+        },
+      },
+      {
+        name: t(getFunctionTooltip(entity)),
+        dataQa: 'status-change',
+        display: isExecutable && isCodeAppsEnabled,
+        disabled: playerStatus === SimpleApplicationStatus.UPDATING,
+        Icon: PlayerContextIcon,
+        iconClassName: classNames({
+          ['text-error']: playerStatus === SimpleApplicationStatus.UNDEPLOY,
+          ['text-accent-secondary']:
+            playerStatus === SimpleApplicationStatus.DEPLOY,
+          ['animate-spin-steps']:
+            playerStatus === SimpleApplicationStatus.UPDATING,
+        }),
```

Comment:
```suggestion
        iconClassName: PlayerContextIconClasses[playerStatus]
```
where `PlayerContextIconClasses` is constant outside this component (can be reused in `TalkToCard.tsx`)
```tsx
const PlayerContextIconClasses = {
  [SimpleApplicationStatus.DEPLOY]: 'text-accent-secondary',
  [SimpleApplicationStatus.UNDEPLOY]: 'text-error',
  [SimpleApplicationStatus.UPDATING]: 'animate-spin-steps',
};
```